### PR TITLE
Use new API when no timestamp is set

### PR DESCRIPTION
**Pull request by urielsalis**
### Issue

Due to https://bugs.mojang.com/browse/WEB-1100, the tool was reporting all users as paid and migrated.

### Solution

- Switched to new [API](https://wiki.vg/Mojang_API#Playernames_-.3E_UUIDs)
- Minor refactors to improve readability
- Names is no longer called if the account is legacy, so responses are a bit faster

### Notes

Example CURL request: curl -X POST -d '["fury_fingers16","urielsalis","turtlehunter2"]' -H "Content-Type: application/json" -v -i 'https://api.mojang.com/profiles/minecraft'

First one should show as legacy, second one as paid and migrated, last one should not be there

As the new API doesnt support timestamps to get the history, the legacy API still needs to be called in those cases. Returning the previous name for turtlehunter2 for example

### CODE REVIEW

File: src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java [hunk 1]
```diff
@@ -63,94 +58,136 @@ public void runEvent(CommandEvent event) {
...
+        }
+        return output.toString();
+    }
+
+    private AccountStatus getAccountStatus(String username) {
         try {
-            String str = "https://api.mojang.com/users/profiles/minecraft/" + username;
-            if (timestamp != null) {
-                str = str + "?at=" + timestamp.toString();
+            URL url = new URL("https://api.mojang.com/profiles/minecraft");
```
[the remark this answers is not in the document]
Comment (urielsalis) [reply src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java]: its not autocloseable, disconnecting instead

File: src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java [hunk 2]
```diff
@@ -63,94 +58,136 @@ public void runEvent(CommandEvent event) {
...
+
+            return new AccountStatus(!demo, !legacy, id, name);
+        } catch (IOException e) {
+            getLogger().log(Level.WARNING, "Error looking up player name " + username, e);
+            return new AccountStatus();
+        }
+    }
+
+    private AccountStatus getLegacyAccountStatus(String username, long timestamp) {
+        try {
```
[the remark this answers is not in the document]
Comment (urielsalis) [reply src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java]: its not autocloseable, disconnecting instead

File: src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java [hunk 3]
```diff
@@ -63,94 +58,136 @@ public void runEvent(CommandEvent event) {
...
+                    output.append(String.format("\nName history too long for display: https://namemc.com/profile/%s", accountStatus.id));
+                } else {
+                    output.append(nameHistory);
+                }
+            }
+        }
+        return output.toString();
+    }
+
+    private AccountStatus getAccountStatus(String username) {
```
[the remark this answers is not in the document]
Comment (urielsalis) [reply src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java]: Using optional instead

File: src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java [hunk 4]
```diff
@@ -63,94 +58,136 @@ public void runEvent(CommandEvent event) {
...
+            String name = user.get("name").getAsString();
+
+            return new AccountStatus(!demo, !legacy, id, name);
+        } catch (IOException e) {
+            getLogger().log(Level.WARNING, "Error looking up player name " + username, e);
+            return new AccountStatus();
+        }
+    }
+
+    private AccountStatus getLegacyAccountStatus(String username, long timestamp) {
```
[the remark this answers is not in the document]
Comment (urielsalis) [reply src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java]: Using optional now

File: src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LordRalex) [reply src/main/java/net/ae97/pokebot/extensions/mcnames/McNamesExtension.java]: Refer to https://github.com/AE97/PircBotY-Extensions/blob/master/src/main/java/net/ae97/pokebot/extensions/scrolls/PlayerCommand.java#L59 on proper handling of this

**Comment by urielsalis** [comment]
@LordRalex everything is fixed

**Pull request merged**